### PR TITLE
Support collecting sample rows for SQL transform & Introduce wikisql benchmark

### PR DESCRIPTION
**Pull request by gengliangwang**
* Support collecting sample rows for SQL transform. After the changes, the generated SQL query is more precise on the filter column/values. 
* Introduce wikisql benchmark https://github.com/salesforce/WikiSQL. The benchmark is to compare query results only. We don't compare the query text since the generated query:
  * may contain multiple selection, while wikisql expects one selected column
  * may contains filter condition such as `like`, `in` while wikisql expects `=`,`<`, `>` only
* This PR contains 99 sample tests and currently 86 of them passed.  We will continue improve the execution
accuracy and introduce more tests.

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @xinrong-meng @gatorsmile

File: benchmark/wikisql/wiki_sql.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                break
+        if not found_match:
+            print("Question: {}".format(question))
+            print("Expected query: {}".format(get_sql_query(table, sql["sel"], sql["agg"], sql["conds"])))
+            print("Actual query: {}".format(query))
+            print("Expected result: {}".format(expected_result))
+            print("Actual result: {}".format(spark_ai_result))
+            print("")
+
+    print(f"Matched {matched} out of {len(results)}")
```
Comment (xinrong-meng) [inline benchmark/wikisql/wiki_sql.py:129]: nit: Shall we add the accuracy percentage for easier comparison? Fine to do it as a follow-up though.

Comment (gengliangwang) [reply benchmark/wikisql/wiki_sql.py:129]: I plan to make this an end-to-end test in the future. But let's keep it simple for now.

File: benchmark/wikisql/wiki_sql.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import re
+from argparse import ArgumentParser
+
+from babel.numbers import parse_decimal, NumberFormatError
+from pyspark.sql import SparkSession
+
+from pyspark_ai import SparkAI
+
+
+def generate_sql_statements(table_file):
```
Comment (xinrong-meng) [inline benchmark/wikisql/wiki_sql.py:11]: nit: Shall we add a docstring to call out the sqls generated are to represent tables as views?

File: pyspark_ai/pyspark_ai.py [hunk 3]
```diff
@@ -327,14 +333,42 @@ def create_df(
...
+
+    def _get_sample_spark_rows(self, df: DataFrame, temp_view_name: str) -> str:
+        if self._sample_rows_in_table_info <= 0:
+            return ""
+        columns_str = "\t".join(list(map(lambda f: f.name, df.schema.fields)))
+        try:
+            sample_rows = self._get_dataframe_results(df.limit(3))
+            # save the sample rows in string format
+            sample_rows_str = "\n".join(["\t".join(row) for row in sample_rows])
+        except Exception:
```
Comment (xinrong-meng) [inline pyspark_ai/pyspark_ai.py:361]: Do we want sampling to fail silently?

File: pyspark_ai/pyspark_ai.py [hunk 4]
```diff
@@ -327,14 +333,42 @@ def create_df(
...
+    def _convert_row_as_tuple(self, row: Row) -> tuple:
+        return tuple(map(str, row.asDict().values()))
+
+    def _get_dataframe_results(self, df: DataFrame) -> list:
+        return list(map(self._convert_row_as_tuple, df.collect()))
+
+    def _get_sample_spark_rows(self, df: DataFrame, temp_view_name: str) -> str:
+        if self._sample_rows_in_table_info <= 0:
+            return ""
+        columns_str = "\t".join(list(map(lambda f: f.name, df.schema.fields)))
```
Comment (xinrong-meng) [inline pyspark_ai/pyspark_ai.py:356]: nit: `columns_str = "\t".join([f.name for f in df.schema.fields])`

**Comment by xinrong-meng** [comment]
LGTM with nits and a question, thank you!

**Review by xinrong-meng — APPROVED** [review]

File: benchmark/wikisql/wiki_sql.py (hunk 2 above)
Comment (gengliangwang) [reply benchmark/wikisql/wiki_sql.py]: Sure, updated

File: pyspark_ai/pyspark_ai.py (hunk 3 above)
Comment (gengliangwang) [reply pyspark_ai/pyspark_ai.py:361]: Yes, sample rows are optional

File: pyspark_ai/pyspark_ai.py (hunk 4 above)
Comment (gengliangwang) [reply pyspark_ai/pyspark_ai.py]: Thanks, updated

**Pull request merged**